### DISCUSSION TITLE
Json files staying in download directory

### DISCUSSION THREAD

**Discussion by NateEverden** [philosowaffle/peloton-to-garmin#211] [Q&A] [answered]
so, it doesn't seem like my json files are getting cleaned up.  This means next time I run p2g it will convert them again and try to upload again.

Something else thats weird is I am seeing fit files in my working dir and upload directory,  I have SaveLocalCopy set to false.

**Comment by philosowaffle** [comment]
My best guess is maybe there is some sort of permissions issue, and P2G is not being allowed to delete the files by the filesystem.  If its failing to delete for some reason there is probably an error message in the log [like this](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Common/FileHandling.cs#L114).

Probably need more info about your setup and logs though to give a better guess.

**Reply by philosowaffle** [reply]
Got some more information on this. Issue can be tracked here: #220

**Comment by NateEverden** [comment]
I am running from code (lastest code) from MacOS.  I am not seeing anything in the logs.  Here is what it looks like:

[log20220204.txt](https://github.com/philosowaffle/peloton-to-garmin/files/8015365/log20220204.txt)
[log20220206.txt](https://github.com/philosowaffle/peloton-to-garmin/files/8015366/log20220206.txt)

You can see in the 2nd log it tried to upload the activity from the first log again.


Here is what my config looks like:

{
  "App": {
    "OutputDirectory": "./output",
    "WorkingDirectory": "./working",
    "SyncHistoryDbPath": "./syncHistory.json",
    "EnablePolling": false,
    "PollingIntervalSeconds": 86400,
    "CloseWindowOnFinish": true
  },

  "Format": {
    "Fit": true,
    "Json": false,
    "Tcx": false,
    "SaveLocalCopy": false,
    "IncludeTimeInHRZones": false,
    "IncludeTimeInPowerZones": false,
    "DeviceInfoPath": "./deviceInfo.xml"
  },

  "Peloton": {
    "Email": "notmyemail",
    "Password": "notmypass",
    "NumWorkoutsToDownload": 10,
    "ExcludeWorkoutTypes": [ "meditation" ]
  },

  "Garmin": {
    "Email": "notmyemail",
    "Password": "notmypass",
    "Upload": true,
    "FormatToUpload": "fit",
    "UploadStrategy": 2
  },

  "Observability": {

    "Prometheus": {
      "Enabled": false,
      "Port": 4000
    },

    "Jaeger": {
      "Enabled": false,
      "AgentHost": "localhost",
      "AgentPort": 6831
    },

    "Serilog": {
      "Using": [ "Serilog.Sinks.Console", "Serilog.Sinks.File" ],
      "MinimumLevel": "Information",
      "WriteTo": [
        { "Name": "Console" },
        {
          "Name": "File",
          "Args": {
            "path": "./output/log.txt",
            "rollingInterval": "Day",
            "retainedFileCountLimit": 7
          }
        }
      ]
    }
  }
}

**Reply by philosowaffle** [reply]
"latest code" as in you pulled down the code after Saturday? Because a fix for this was added on Saturday: #220

**Comment by NateEverden** [comment, marked as answer]
Guess I wasnt running the latest code.  Im still learning GIT.  I thought git log would show me if anything changed.  I guess I need to just git pull

**Reply by philosowaffle** [reply]
I think `git log` just prints out the commit history of the local code.  A couple handy commands that should cover most cases:

* `git fetch` - will check in with the server to see if anything new is out there (it does not pull down nor push any changes), very counterintuitive
* `git status` - After the latest info has been fetched from the server, this will report if there are any differences between local and server
* `git pull` - pulls down changes from server

Running `git status` again after `git pull` should confirm that there are no longer any differences between local/remote.

It's the `git fetch` command that is key since sometimes git is "lazy" about checking the server for updates.
